### PR TITLE
fix cell + delete cell.dm

### PR DESCRIPTION
**Pull request by Xander3359**
## About The Pull Request
Fix recharger cell not updating ammo count when it recharges
Fix having to remove the recharger cell and putting it back in the gun before being able to shoot again.
Delete cell.dm, the reason for this is because I find it very weird having to look for the /cell procs and it's an a completely different file
## Why It's Good For The Game
Bugfix
## Changelog
:cl:
fix: Recharger cells now update ammo count when it recharges
fix: Recharger cells no longer break once they hit 0 ammo
code: delete cell.dm, merges the code into power_cells.dm
/:cl:

### CODE REVIEW

File: code/game/objects/items/power_cells.dm [hunk 1]
```diff
@@ -13,17 +13,250 @@
...
+			if (prob(25))
+				qdel(src)
+				return
+			if (prob(25))
+				corrupt()
+		if(EXPLODE_WEAK)
+			if (prob(25))
+				corrupt()
+
+/obj/item/cell/proc/get_electrocute_damage()
```
Comment (Lumipharon) [inline code/game/objects/items/power_cells.dm:245]: new procs under old procs. also you touched it so you get to autodoc it.

File: code/game/objects/items/power_cells.dm [hunk 2]
```diff
@@ -13,17 +13,250 @@
...
+	/// A tracker for use in self-charging
+	var/last_use = 0
+	/// How long it takes for the cell to start recharging after last use
+	var/charge_delay = 0
 	///used to track what set of overlays to use to display charge level
 	var/charge_overlay = "cell"
 
+// the power cell
+// charge from 0 to 100%
+// fits in APC to provide backup power
```
Comment (Lumipharon) [inline code/game/objects/items/power_cells.dm:37]: kill these random pointless comments while you're at it

File: code/modules/projectiles/gun_system.dm [hunk 3]
```diff
@@ -1344,6 +1344,7 @@
 		update_ammo_count()
 		update_icon()
 		to_chat(user, span_notice("You reload [src] with [new_mag]."))
+		RegisterSignal(new_mag, COMSIG_CELL_SELF_RECHARGE, TYPE_PROC_REF(/obj/item/weapon/gun, update_ammo_count))
```
Comment (Lumipharon) [inline code/modules/projectiles/gun_system.dm:1347]: I told you that you needed to register for emp_act as well.

File: code/game/objects/items/power_cells.dm [hunk 4]
```diff
@@ -13,17 +13,250 @@
...
+
+	QDEL_IN(src, 1)
+
+/obj/item/cell/proc/corrupt()
+	charge /= 2
+	maxcharge /= 2
+	if (prob(10))
+		rigged = 1 //broken batterys are dangerous
+
+/obj/item/cell/emp_act(severity)
```
Comment (Lumipharon) [inline code/game/objects/items/power_cells.dm:220]: as above, needs a sig, and icon update.

**Review by Lumipharon — CHANGES_REQUESTED** [review]

File: code/game/objects/items/power_cells.dm (hunk 1 above)
Comment (Xander3359) [reply code/game/objects/items/power_cells.dm:245]: Fuck. Ok this will take me a bit

File: code/game/objects/items/power_cells.dm [hunk 5]
```diff
@@ -13,21 +13,256 @@
...
+	charge += amount_used
+	return amount_used
+
+///return % charge of cell
+/obj/item/cell/proc/percent()
+	return 100 * (charge / maxcharge)
+
+///Returns TRUE if charge is equal to maxcharge
+/obj/item/cell/proc/fully_charged()
+	return (charge == maxcharge)
```
Comment (XElectricX) [inline code/game/objects/items/power_cells.dm:221]:
```suggestion
/obj/item/cell/proc/is_fully_charged()
	return charge == maxcharge
```
I recommend changing this slightly misleading proc name. Also the parentheses don't have any effect here.

**Pull request merged**